src/declaration.ts

```diff
@@ -37,6 +37,31 @@ export class FunctionDecl<F extends AnyFunction = AnyFunction> extends BaseDecl<
   }
 }
 
+export const isHoistedFunctionDecl = typeGuard("HoistedFunctionDecl");
+
+/**
+ * Instead of serializing this function to AST, this function has been hoisted to a module export with a random name.
+ */
+export class HoistedFunctionDecl<
```

Comment:
How will we determine the dependencies and auto wire environment variables and iam policies? Walk the FunctionDecl like we do for the functionless integrations?

Reply:
Ohh, ha, this needs to be cleaned up, from before using Pulumi.

Reply:
There are a few options here, once is to use the serialize function like Cloudy does, another is to use codegen. I will figure this out today.